lib/rules/prefer-const.js

```diff
@@ -199,6 +200,28 @@ function groupByDestructuring(variables, ignoreReadBeforeAssign) {
         }
     }
 
+    if (!checkingMixedDestructuring) {
+        for (const key of identifierMap.keys()) {
+            const destructureGroup = identifierMap.get(key);
+            const destructureElement = destructureGroup[0];
+            let elementsInHost;
+
+            if (destructureElement && destructureElement.parent) {
+                if (destructureElement.parent.elements) {
+                    elementsInHost = destructureElement.parent.elements;
+                }
+
+                if (destructureElement.parent.parent.type === "ObjectPattern") {
+                    elementsInHost = destructureGroup[0].parent.parent.properties;
+                }
+            }
+
+            if (elementsInHost && elementsInHost.length && elementsInHost.length !== destructureGroup.length) {
```

Comment:
I think this strategy will fail for things like

```js
let foo; [foo, []] = qux;
```

or like this:

```js
let foo, bar; [foo, [bar, baz.qux]] = qux;
```

With the `desctructuring: all` option, the first case should be invalid, and the second case should be valid.

I think this is going to be a problem with the approach of checking the number of elements in a destructuring pattern, because it won't be able to account for nested destructuring.